airbyte-integrations/connectors/source-faker/source_faker/streams.py

```diff
@@ -0,0 +1,190 @@
+
+    for i in range(purchases_count, total_user_records):
+        purchases = self.generate_purchases(i + 1, purchases_count)
+        for purchase in purchases:
+            yield purchase
+            purchases_count += 1
+
+    self.state = {self.cursor_field: purchases_count, "seed": self.seed}
+
+class Users(Stream, IncrementalMixin):
+  primary_key = None
+  cursor_field = "id"
+
+  def __init__(self, count: int, seed: int, records_per_sync: int, records_per_slice: int, **kwargs):
+    super().__init__(**kwargs)
+    self._cursor_value = -1
+    self.count = count
+    self.seed = seed
+    self.records_per_sync = records_per_sync
+    self.records_per_slice = records_per_slice
+    self.person = Person(locale=Locale.EN, seed=self.seed)
+    self.dt = Datetime(seed=self.seed)
+
+  @property
+  def state_checkpoint_interval(self) -> Optional[int]:
+     return self.records_per_slice
+
+  @property
+  def state(self) -> Mapping[str, Any]:
+    if hasattr(self, '_state'):
+      return self._state
+    else:
+      return {self.cursor_field: 0}
+
+  @state.setter
+  def state(self, value: Mapping[str, Any]):
+    self._state = value
+
+  def generate_user(self, user_id: int):
+    time_a = self.dt.datetime()
+    time_b = self.dt.datetime()
+
+    profile = {
+        "id": user_id + 1,
+        "created_at": format_airbyte_time(time_a if time_a <= time_b else time_b),
+        "updated_at": format_airbyte_time(time_a if time_a > time_b else time_b),
+        "name": self.person.name(),
+        "title": self.person.title(),
+        "age": self.person.age(),
+        "email": self.person.email(),
+        "telephone": self.person.telephone(),
+        "gender": self.person.gender(),
+        "language": self.person.language(),
+        "academic_degree": self.person.academic_degree(),
+        "nationality": self.person.nationality(),
+        "occupation": self.person.occupation(),
+        "height": self.person.height(),
+        "blood_type": self.person.blood_type(),
+        "weight": self.person.weight(),
+    }
+
+    while not profile["created_at"]:
+        profile["created_at"] = format_airbyte_time(self.dt.datetime())
+
+    if not profile["updated_at"]:
+        profile["updated_at"] = profile["created_at"] + 1
+
+    return profile
+
+  def read_records(self, sync_mode: SyncMode, cursor_field: List[str] = None, stream_slice: Mapping[str, Any] = None, stream_state: Mapping[str, Any] = None) -> Iterable[Mapping[str, Any]]:
+    total_records = self.state[self.cursor_field] if hasattr(self.state, self.cursor_field) else 0
+    records_in_sync = 0
+
+    for i in range(total_records, self.count):
+        user = self.generate_user(i)
+        yield user
+        total_records += 1
+        records_in_sync += 1
+
+        if records_in_sync == self.records_per_sync:
+            break
+
+    self.state = {self.cursor_field: total_records, "seed": self.seed}
+    set_total_user_records(total_records)
```

Comment:
@sherifnada setting `self.state = {...}` still doesn't seem to emit a state message.  Any tips?  I'm testing for this explicitly (`unit_tests.py` line 83)